### PR TITLE
Add support for Android editor builds for Meta devices

### PR DESCRIPTION
**Pull request by m4gr3d**
Requires https://github.com/godotengine/godot/pull/96624

### CODE REVIEW

**Review by dsnopek — APPROVED** [review]
I didn't test it, but the gradle command and the build artifact paths look good to me.

**Review by BastiaanOlij — APPROVED** [review]
LGTM

File: build-android/build.sh [hunk 1]
```diff
@@ -35,17 +35,24 @@ if [ "${CLASSICAL}" == "1" ]; then
   $SCONS platform=android arch=x86_64 $OPTIONS target=editor store_release=${store_release}
 
   pushd platform/android/java
+  # Generate the regular Android editor
   ./gradlew generateGodotEditor
+  # Generate the Android editor for Meta devices
```
Comment (akien-mga) [inline build-android/build.sh:40]:
```suggestion
  # Generate the regular Android editor.
  ./gradlew generateGodotEditor
  # Generate the Android editor for Meta devices.
```

File: build-android/build.sh [hunk 2]
```diff
@@ -35,17 +35,24 @@ if [ "${CLASSICAL}" == "1" ]; then
...
   popd
 
   mkdir -p /root/out/tools
   # Copy the generated Android editor binaries (apk & aab).
   if [ "$store_release" == "yes" ]; then
-    cp bin/android_editor_builds/android_editor-release.apk /root/out/tools/android_editor.apk
-    cp bin/android_editor_builds/android_editor-release.aab /root/out/tools/android_editor.aab
+    cp bin/android_editor_builds/android_editor-google-release.apk /root/out/tools/android_editor.apk
+    cp bin/android_editor_builds/android_editor-google-release.aab /root/out/tools/android_editor.aab
+    # For the Meta build, we only copy the apk
```
Comment (akien-mga) [inline build-android/build.sh:49]:
```suggestion
    # For the Meta build, we only copy the apk.
```

File: build-android/build.sh [hunk 3]
```diff
@@ -35,17 +35,24 @@ if [ "${CLASSICAL}" == "1" ]; then
...
+    cp bin/android_editor_builds/android_editor-meta-release.apk /root/out/tools/android_editor_meta.apk
   else
-    cp bin/android_editor_builds/android_editor-debug.apk /root/out/tools/android_editor.apk
-    cp bin/android_editor_builds/android_editor-debug.aab /root/out/tools/android_editor.aab
+    cp bin/android_editor_builds/android_editor-google-debug.apk /root/out/tools/android_editor.apk
+    cp bin/android_editor_builds/android_editor-google-debug.aab /root/out/tools/android_editor.aab
+    # For the Meta build, we only copy the apk
```
Comment (akien-mga) [inline build-android/build.sh:54]:
```suggestion
    # For the Meta build, we only copy the apk.
```

**Review by akien-mga — APPROVED** [review]
Seems fine, but I think I'll rework this further to add more configuration options for this script.

It's predominantly meant for our official builds, and thus fine as is, but a lot of people also try to use it to make their own builds of Godot, and they won't care to build the Android editor for either Google Play or Horizon Store IMO, so we can make it optional.

This would help improve the usability around the whole `store_release` and keystore config, which can be hard to get right for those users.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**